# Hypothesis Testing Q2

In [2]:
import pandas as pd
import numpy as np
from scipy import stats

# Data description


α == 0.05 (95% Confidence)

Y == Continious
X == Discrete

Is Y1, Y2, Y3 and Y4 normal ?

H0 = Y1, Y2, Y3 and Y4 are normal
H1 = Y1, Y2, Y3 and Y4 are not normal

# Normality Test

In [3]:
df = pd.read_csv(r"C:\Users\Shivam\Downloads/LabTAT.csv")


In [4]:
stats.shapiro(df["Laboratory 1"])

ShapiroResult(statistic=0.9901824593544006, pvalue=0.5506953597068787)

In [6]:
stats.shapiro(df["Laboratory 2"])

ShapiroResult(statistic=0.9936322569847107, pvalue=0.8637524843215942)

In [7]:
stats.shapiro(df["Laboratory 3"])

ShapiroResult(statistic=0.9886345267295837, pvalue=0.4205053448677063)

In [8]:
stats.shapiro(df["Laboratory 4"])

ShapiroResult(statistic=0.9913753271102905, pvalue=0.6618951559066772)

(0.9901824593544006, 0.5506953597068787)
P value for Laboratory 1 == 0.55 > α


(0.9936322569847107, 0.8637524843215942)
P value for Laboratory 2 == 0.86 > α


(0.9886345267295837, 0.4205053448677063)
P value for Laboratory 3 == 0.42 > α


(0.9913753271102905, 0.6618951559066772)
P value for Y1 (Unit B) == 0.66 > α

HO is accepted. Thats is Y1, Y2, Y3, and Y4 are normal

# Variance

In [9]:
stats.levene(df["Laboratory 1"], df["Laboratory 2"], df["Laboratory 3"], df["Laboratory 4"])

LeveneResult(statistic=2.599642500418024, pvalue=0.05161343808309816)

P Value of Variance test is == 0.051 > α

H0 is accepted
Thus we will perform one way Anova Test

# Model

In [10]:
stats.stats.f_oneway(df["Laboratory 1"], df["Laboratory 2"], df["Laboratory 3"], df["Laboratory 4"])

F_onewayResult(statistic=118.70421654401437, pvalue=2.1156708949992414e-57)

P value of the One way Anova test is == 2e-16 < α

Thus H1 is accepted.
Mean TAT for 4 laboratories not equal (There is a significance difference between TAT of the laboratories)